crates/services/p2p/src/discovery/discovery_config.rs

```diff
@@ -97,7 +102,7 @@ impl Config {
         self
     }
 
-    pub fn finish(self) -> Behaviour {
+    pub async fn finish(self) -> anyhow::Result<Behaviour> {
```

Comment:
Just curious, have you considered implementation that doesn't require `async` during construction? Maybe it is possible to resolved addresses in the `fn start`?

Reply:
do you mean here - https://github.com/FuelLabs/fuel-core/blob/1d1560f2fd05b2d28f7df14c98bf65d98fbbcf53/crates/services/p2p/src/p2p_service.rs#L238 ?

Reply:
addressed in 5b4671f1f